system_setup/server/controllers/configure.py

```diff
@@ -143,29 +143,35 @@ async def _install_check_lang_support_packages(self, lang, env) -> bool:
             return False
 
         packages = [pkg for pkg in cp.stdout.strip().split(' ') if pkg]
-        if len(packages) == 0:
-            log.debug("%s didn't recommend any packages. Nothing to do.",
-                      clsCommand)
-            return True
-
         cache = apt.Cache()
         if self.app.opts.dry_run:
             packs_dir = os.path.join(self.model.root,
                                      apt_pkg.config
                                      .find_dir("Dir::Cache::Archives")[1:])
             os.makedirs(packs_dir, exist_ok=True)
             try:
+                if len(packages) == 0:
+                    message = "{} didn't recommend any packages." \
+                              " Nothing to do.".format(clsCommand)
+                    log.debug(message)
+                    msgFile = os.path.join(packs_dir, "msgFile")
```

Comment:
Thanks for looking into this.

Note that dry run is used outside of the integration tests - in particular ubuntu-desktop-installer uses it heavily for devel purposes.

While I appreciate that this should solve the problem, I wonder if we should consider another route.
What if we added a mock `check-language-support` to the integration tests that has output you control?  This could also mean that you could do a more calculated test - not just check that the cache is present or absent, but check for specific files to be present.  That would also let you test the no-op case.  What do you think?

Reply:
I think @dbungert's strategy is the best one. Wdyt @CarlosNihelton?

Reply:
Sure! I'll take care of that. Thanks for the recommendations.